lib/arel/sql_to_arel/pg_query_visitor.rb

```diff
@@ -719,7 +719,9 @@ def visit_SelectStmt(
         select_core.havings = [visit(having_clause)] if having_clause
         select_core.windows = visit(window_clause) if window_clause
         select_core.into = visit(into_clause) if into_clause
-        select_core.top = ::Arel::Nodes::Top.new visit(limit_count) if limit_count
+        if (ActiveRecord.version.version.to_i != 6) && limit_count
```

Comment:
Why is this change necessary? If this is because `::Arel::Nodes::Top` does not exist, you might as well define it in `lib/arel/extensions/top.rb`

Reply:
This block is necessary for active_record 5.2, without it Arel tree has a different structure (Arel tree includes `Arel::Nodes::Top` while it should be nil). Please, see the two arel trees in the screenshot

![image](https://user-images.githubusercontent.com/6542167/98356874-27a73080-2046-11eb-85cf-00519ddeb508.png)

or run the spec `rspec ./spec/arel/sql_to_arel_spec.rb:874`

Reply:
Yeah I get you, but I'm saying we can keep the Arel::Nodes::Top node so we don't need this conditional branch :).

Reply:
Got it! thank you for the suggestion, @mvgijssel

Reply:
I tried to define `Top` in `lib/arel/extensions/top.rb`
```
# frozen_string_literal: true

module Arel # :nodoc: all
  module Nodes
    class Top < Unary
    end
  end
end
```
As result, all specs passed for ActiveRecord 5.2 but some specs failed for ActiveRecord 6.0 with an error `message: uninitialized constant Arel::Nodes::Top`.

@mvgijssel  Given the time pressure, I would like to go like it is now if you don't have objection?

Reply:
The issue was fixed after adding reference to file. @mvgijssel thanks for the suggestion!